pandas/tests/test_base.py

```diff
@@ -296,14 +296,21 @@ def test_none_comparison(self):
                 # result = None != o  # noqa
                 # assert result.iat[0]
                 # assert result.iat[1]
+                if (is_datetime64_dtype(o) or is_datetimetz(o)):
```

Comment:
this might be simpler to separate out into datetimelike and non-datetimelike

Reply:
as in separate tests?

Reply:
yes, all of these ifs make it hard to read

Reply:
OK.  This whole test should be parametrized, and there `isinstance(o, Series)` check should be rendered unnecessary.  Vote for doing this in the next pass.

Reply:
ok that's fine